src/Aspire.Hosting/Backchannel/AppHostRpcTarget.cs

```diff
@@ -0,0 +1,69 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Dashboard;
+using Aspire.Hosting.Devcontainers.Codespaces;
+using Aspire.Hosting.Utils;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Aspire.Hosting.Backchannel;
+
+internal class AppHostRpcTarget(ILogger<AppHostRpcTarget> logger, ResourceNotificationService resourceNotificationService, IServiceProvider serviceProvider)
+{   
+    public async IAsyncEnumerable<(string Resource, string Type, string State, string[] Endpoints)> GetResourceStatesAsync([EnumeratorCancellation]CancellationToken cancellationToken)
+    {
+        var resourceEvents = resourceNotificationService.WatchAsync(cancellationToken);
+
+        await foreach (var resourceEvent in resourceEvents.WithCancellation(cancellationToken).ConfigureAwait(false))
+        {
+            if (resourceEvent.Resource.Name == "aspire-dashboard")
```

Comment:
Maybe also skip hidden resources?

Reply:
Will address this in follow-up PR. I also need to deal with health status better and replicas.